### PR TITLE
Update IR ExecutorOutput message to use ArtifactList for artifacts field.

### PR DESCRIPTION
**Pull request by chensun**

### CODE REVIEW

File: api/v2alpha1/pipeline_spec.proto [hunk 1]
```diff
@@ -338,6 +338,12 @@ message RuntimeArtifact {
   map<string, Value> custom_properties = 5;
 }
 
+// Message that represents a list of artifacts.
+message ArtifactList {
+  // A list of parameters.
```
Comment (Bobgy) [inline api/v2alpha1/pipeline_spec.proto:343]: nit: is `parameters` a typo?

Comment (chensun) [reply api/v2alpha1/pipeline_spec.proto]: Good catch! fixed.

**Comment by numerology** [comment]
/lgtm
/approve

Thanks!

**Comment by Bobgy** [comment]
Can you link me to why one key can have multiple artifacts? Is it related to directory artifact?
/approve

**Comment by numerology** [comment]
> Can you link me to why one key can have multiple artifacts? Is it related to directory artifact?

Semantically, when the execution mode is async, it's possible to consume > 1 artifact in one execution. For example, a training job can consume multiple batches of examples produced by upstream tasks.

**Pull request merged**